server/src/main/scala/com/cloudera/livy/utils/SparkYarnApp.scala

```diff
@@ -78,7 +78,8 @@ class SparkYarnApp private[utils] (
   private var yarnDiagnostics: IndexedSeq[String] = IndexedSeq.empty[String]
 
   override def log(): IndexedSeq[String] =
-    process.map(_.inputLines).getOrElse(ArrayBuffer.empty[String]) ++ yarnDiagnostics
+    process.map(_.inputLines).getOrElse(ArrayBuffer.empty[String]) ++ yarnDiagnostics ++
```

Comment:
I think we should print yarnDiagnostics after errorLines because it's changing. Can we also add some formatting to separate logs from stdout/stderr/yarn?